ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java

```diff
@@ -407,8 +408,9 @@ public boolean shouldRun(TimerImpl timer, TransactionManager tm) {
                     statement.setTimestamp(6, timestamp(timer.getNextExpiration()));
                 }
             } catch (SQLException e) {
-                // something wrong with the preparation
-                throw new RuntimeException(e);
+                // fix for WFLY-10130
+                EjbLogger.EJB3_TIMER_LOGGER.exceptionCheckingIfTimerShouldRun(timer, e);
```

Comment:
This whole catch block should just be deleted, and the exception should propagate to the outer catch block